### DISCUSSION TITLE
Installation and usage without Magisk (on LineageOS)?

### DISCUSSION THREAD

**Discussion by onli** [VR-25/acc#186] [Q&A]
On LOS, the only available root solution was Magisk. But Magisk breaks currently on every upgrade when using their only recommended installation method, see https://github.com/topjohnwu/Magisk/issues/3820, so basically it does not exist in a usable state. 

So (ignoring the option to .zip flash Magisk anyway) the only tool at hand is the recovery, adb, and adb root (right?). Is it possible to install and use acc in such an environment? Take the latest release, sideload it? Or use adb root to run the install script on the device?

I understand that acc would only run with its default configuration, but that might be okay, especially since that configuration could be changed during the installation. 

I saw the install section in the Readme, but the possibility of this "rootless" - or rather, Magiskless - usage stayed unclear to me.

**Comment by VR-25** [comment]
ACC can work under adb root. One can even modify the config (`/data/adb/vr25/acc-data/config.txt`), as long as it's done with root privileges.
The only major limitation, in this context, is that it won't autostart on boot (unless there's an autorun script that initializes it as root). That said, it has to be manually initialized (with adb root), by running `/data/adb/vr25/acc/service.sh`.
In addition, since acc is always systemless, one cannot run regular commands, unless `/sbin/` is available and writable. When that's not the case, instead of running, for instance, `acc -i`, one would have to run `/dev/acc -i` instead.

To install, simply extract the zip and run `install.sh` on terminal, as root. The daemon starts automatically after installation. However, as noted above, it must be initialized manually after each boot, unless there's some sort of  init.d script to automate that.

**Reply by arooni** [reply]
I too am on lineage (18.1) on a pixel xl (marlin), however, I was attempting to use with https://github.com/MatteCarra/AccA ;; which did not seem to work in limiting charge.  I can run your tool via termux and root, but not sure how to go about a standard limiting charge to 80 even when plugged in, and to default to using the power cable above 80%

EDIT: just ran 

sudo acc 80 

and after running the test switches, I see that its current is -272 mA (Discharging) so I *think* this means it won't start charging till 75?  

oh bummer, look like percentage went up from 98 to 99%.  Does this require a custom kernel of some sort?

**Reply by VR-25** [reply]
https://github.com/VR-25/acc/files/15137601/acc_v2024.4.26-rc_202404260_0003.zip

This fixes compatibility issues with devices that have unconventional/weird battery interfaces.
